### PR TITLE
Places Greater Importance on Justifying Your Changes in the Why Its Good for the Game Section

### PR DESCRIPTION
**Pull request by Kylerace**
## About The Pull Request
This PR: 
1. Moves the part explaining why you should justify your changes to its own section in contributing.md, and expands on it greatly. 
2. Adjusts the pull request template to place a further emphasis on arguing for why your changes are good in the Why It's Good For The Game section.

## Why It's Good For The Game
Changes that piss people off because of x, y, and z can have merit in spite of that, but that merit should be explained. Period. There is no downside to putting in a reasonable effort to justify why what you're doing will improve the game. 

Lazy, low effort explanations for the merits of your changes, wholly unsuited to the level of justification needed to be considered adequate by anyone, should no longer be accepted into our codebase until they are at the very least adequate. This isn't stuff like bugfixes or tiny features no one has any problems with, this is for stuff like #58038 or #62947 or #57831 or #64693 which have either abysmal justification for their changes or are outright refusing any real attempt at a justification relative to their effects.

Proper justification for controversial PRs should be required even if you have explicit support of a maintainer, or maintainers, or a head coder, or multiple head coders, or the entirety of humanity, or God himself before the pr is even made. Because when a PR is merged, the argument for its changes become the codebases argument instead of just the authors. Thus all potentially controversial changes should have a level of effort put into their arguments proportional to their effect on the game.

### CODE REVIEW

**Comment by Kylerace** [comment]
@tgstation/commit-access

**Comment by tralezab** [comment]
less trolling the playerbase is good. there's a mindset of "I have maintainer backing I can just shit on players now" that is very unhealthy for development

**Comment by SmArtKar** [comment]
Why It's Good For The Game is too small, I feel that you haven't explain why this is good for us and community in general
image
Jokes aside this should've been done much sooner as certain people had made PRs just because of their hatred for certain jobs. Gud stuff.

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -130,14 +130,24 @@ There is no strict process when it comes to merging pull requests. Pull requests
...
 
 * After leaving reviews on an open pull request, maintainers may convert it to a draft. Once you have addressed all their comments to the best of your ability, feel free to mark the pull as `Ready for Review` again.
 
+## Justifying Your Changes
+You must explain why you are submitting the pull request in the "Why It's Good For The Game" section of your pull request, and how you think your change will be beneficial to the game. Failure to do so will be grounds for rejecting your PR wholesale, or requiring that you fix it before it is merged. A reasonable justification for your changes is a requirement. 
+
+Your "Why It's Good For The Game" section must make a good faith and reasonable attempt to:
+* Convince the community that the current state of affairs in the game is not good, and needs changing.
```
Comment (LemonInTheDark) [inline .github/CONTRIBUTING.md:143]: > community

mildly cringe implications, understandable but still cringe

Comment (Kylerace) [reply .github/CONTRIBUTING.md:143]: _**A B S O L U T E**_

Comment (LemonInTheDark) [reply .github/CONTRIBUTING.md:143]: I think it's important less to convince people, and more to leave a record of your intent.
Kind of in the same vein as hackmds, that's what bothers me the most when reading old prs, changes with no explanation on linked understanding.

**Comment by LemonInTheDark** [comment]
Like the effect, don't like the implication that the general community's opinion is a deciding factor

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (Kylerace) [reply .github/CONTRIBUTING.md]: changed "convince the community" to "argue and assert"

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 ## Why It's Good For The Game
 
-<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->
+<!-- Argue why these changes would benefit the game to the community, especially if they are controversial and/or far reaching. If you can't justify it in words, it might not be worth adding. -->
```
Comment (LemonInTheDark) [inline .github/PULL_REQUEST_TEMPLATE.md:10]: applies here too

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -130,14 +130,24 @@ There is no strict process when it comes to merging pull requests. Pull requests
...
+* Assert and argue that the current state of affairs in the game is not good, and needs changing.
+* Assert and argue that your pull request will either fix or help fix the problems you described.
+* Assert and argue that any downsides introduced by your solution as a matter of design, are worth it, and why they are worth it.
+
+More controversial changes have higher standards for justification to be considered reasonable. A bugfix does not typically require any effort at all in justification as its worth is usually self evident, however a major feature overhaul or balance change may require significant argumentation to adequately justify its supposed benefit to the game.
+
+This does not change if your pull request is supported and/or requested by maintainers before it is opened. This does not change if your pull request is supported and/or requested by head coders before it is opened. This does not change if your pull request has a design document that justifies your changes inside it. You must always properly justify changes (that need justification) within the pull request, even if you also do it elsewhere.
+
```
Comment (LemonInTheDark) [inline .github/CONTRIBUTING.md:150]: Should give some of the why here

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (Kylerace) [reply .github/PULL_REQUEST_TEMPLATE.md]: changed to:
`Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place.`

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (Kylerace) [reply .github/CONTRIBUTING.md]: changed to 
`
This is still a requirement if your pull request is supported and/or requested by maintainers before it is opened. This is still a requirement if your pull request is supported and/or requested by head coders before it is opened. The purpose of arguing for your changes is not to convince just the maintainer team of its merits, it is to document the "why" behind your changes to the game to a necessary level of detail. The reason behind a change must exist as it is the purpose of this codebase to improve the game, thus said reasoning must be adequately stated and explained.

This is also still a requirement if your pull request has a corresponding design document that justifies your changes inside it. You must always properly justify changes (those that actually need justification) within the pull request, even if you also do it elsewhere. This is to ensure that:
1. All reviewers can easily see the reasoning behind your changes on the pull request itself, no reliance on other sites required.
2. The actual, manifested implementation of the idea behind the design document is being justified after said implementation is actually realized. This is in contrast to any reasoning put on the design document itself, which very well may have been made before any work was done on it, possibly even by an author different from the author of the pull request. Any idea in the design document may have had compromises put into it due to complications not seen in the original vision, thus the current state of the implementation (the pull request as it stands) must be defended, explained, and ultimately justified in and of itself. Of course, you should still list the design document the pull request is implementing, and may even use arguments from the design document if said arguments are applicable to the current reality of your proposed changes.
`

**Comment by Ghilker** [comment]
Love it how 3 out of 4 are mine

**Comment by Kylerace** [comment]
justify your changes

**Comment by Shadowflame909** [comment]
> Love it how 3 out of 4 are mine a7634Nl2_700w_0

with all the pressure rising you cause in the playerbase. You could make a plasma-bomb! get it...because you mainly code for atmos.

**Comment by Lauwma** [comment]
It shouldn't be "why it's good for the game" but "why it's good for the players"

**Comment by Mothblocks** [comment]
That is the same thing to everyone

**Comment by wesoda25** [comment]
> Love it how 3 out of 4 are mine 

He was so close guys!

**Pull request merged**